src/components/protocol_handler/test/protocol_handler_tm_test.cc

```diff
@@ -168,7 +171,8 @@ class ProtocolHandlerImplTest : public ::testing::Test {
                                                     std::string("BTMAC")),
                                          connection_id);
   }
-  void AddSession() {
+
+  void AddSession(TestAsyncWaiter* waiter, uint32_t* times) {
```

Comment:
@istoilovgithub , please, use SharedPointer for `TestAsyncWaiter*` and pass `times` by reference